components/gong/actions/get-extensive-data/get-extensive-data.mjs

```diff
@@ -68,20 +138,60 @@ export default {
       app,
       getExtensiveData,
       maxResults,
+      context,
+      contextTiming,
+      includeParties,
+      exposedFieldsContent,
+      exposedFieldsInteraction,
+      includePublicComments,
+      includeMedia,
       ...filter
     } = this;
 
     if (filter?.workspaceId && filter?.callIds) {
       throw new ConfigurationError("Must not provide both `callIds` and `workspaceId`");
     }
 
+    const contentSelector = {
+      "context": context || "None",
+      ...(contextTiming.length > 0 && {
+        "contextTiming": contextTiming,
+      }),
+      "exposedFields": {
+        "parties": includeParties || false,
+        "content": {
+          "structure": exposedFieldsContent.structure || false,
```

Comment:
Users can use custom expressions on inputs that come here as strings, make sure to parse exposedFieldsContent and exposedFieldsInteraction before treating as objects.
You can add the following function in common utils file and use here

`function parseObject(obj) {
  if (!obj) return undefined;

  if (Array.isArray(obj)) {
    return obj.map((item) => {
      if (typeof item === "string") {
        try {
          return JSON.parse(item);
        } catch (e) {
          return item;
        }
      }
      return item;
    });
  }
  if (typeof obj === "string") {
    try {
      return JSON.parse(obj);
    } catch (e) {
      return obj;
    }
  }
  return obj;
};`